modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm

```diff
@@ -1,177 +1,206 @@
+					reducing libido and calming them. Non-habit forming and non-addictive."
 	taste_description = "dull bitterness"
 	taste_mult = 2
 	color = "#D9D9D9"//rgb(157, 157, 157)
 	reagent_state = SOLID
+	life_pref_datum = /datum/preference/toggle/erp/aphro
 
-/datum/reagent/drug/camphor/on_mob_life(mob/living/carbon/human/M)
-	if(M.client && (M.client.prefs.read_preference(/datum/preference/toggle/erp/aphro)))
-		if(ishuman(M))
-			var/old_arousal = M.arousal
-			M.adjustArousal(-12)
-			M.adjustPleasure(-3)
-			if(M.arousal <= 0 && old_arousal > 0)
-				to_chat(M, span_notice("You no longer feel aroused."))
-	..()
-
-//Pentacamphor. Used to PERMANENTLY reduce libido. Possibly can cure bimbofication. I wrote this with a straight face, what am i doing?
+	//amount to reduce arousal by
+	var/arousal_change = -12
+	//amount to reduce pleasure by
+	var/pleasure_change = -3
 
-/datum/reagent/drug/pentacamphor
+/datum/reagent/drug/aphrodisiac/camphor/life_effects(mob/living/carbon/human/exposed_mob)
+	. = ..()
+	var/old_arousal = exposed_mob.arousal
+	exposed_mob.adjustArousal(arousal_change)
+	exposed_mob.adjustPleasure(pleasure_change)
+	if(exposed_mob.arousal <= 0 && old_arousal > 0)
+		to_chat(exposed_mob, span_notice("You no longer feel aroused."))
+
+//Pentacamphor. Used to purge crocin and hexacrocin. Can permanently disable arousal or cure bimbofication on overdose.
+/datum/reagent/drug/aphrodisiac/camphor/pentacamphor
 	name = "Pentacamphor"
 	description = "Chemically condensed camphor. Causes an extreme reduction in libido and a permanent one if overdosed. Non-addictive."
 	taste_description = "tranquil celibacy"
 	color = "#D9D9D9"//rgb(255, 255, 255)
 	reagent_state = SOLID
 	overdose_threshold = 20
+	arousal_change = -18
+	overdose_pref_datum = /datum/preference/toggle/erp/aphro
 
-/datum/reagent/drug/pentacamphor/on_mob_life(mob/living/carbon/human/M)
-	if(M.client && (M.client.prefs.read_preference(/datum/preference/toggle/erp/aphro)))
-		if(ishuman(M))
-			var/old_arousal = M.arousal
-			M.adjustArousal(-18)
-			M.adjustPleasure(-3)
-			if(M.arousal <= 0 && old_arousal > 0)
-				to_chat(M, span_notice("You no longer feel aroused."))
-
-		if(M.reagents.has_reagent(/datum/reagent/drug/crocin))
-			M.reagents.remove_reagent(/datum/reagent/drug/crocin, 20)
-		if(M.reagents.has_reagent(/datum/reagent/drug/hexacrocin))
-			M.reagents.remove_reagent(/datum/reagent/drug/hexacrocin, 20)
-	..()
+/datum/reagent/drug/aphrodisiac/camphor/pentacamphor/life_effects(mob/living/carbon/human/exposed_mob)
+	. = ..()
+	if(exposed_mob.reagents.has_reagent(/datum/reagent/drug/aphrodisiac/crocin))
+		exposed_mob.reagents.remove_reagent(/datum/reagent/drug/aphrodisiac/crocin, 20)
+	if(exposed_mob.reagents.has_reagent(/datum/reagent/drug/aphrodisiac/crocin/hexacrocin))
+		exposed_mob.reagents.remove_reagent(/datum/reagent/drug/aphrodisiac/crocin/hexacrocin, 20)
```

Comment:
I feel like those numbers shouldn't be just random numbers and should be vars instead.